File: Anyelo Carrillo Java/cajero.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class Usuario {
    String nombre;
    int pin;
    int saldo;

    Usuario(String nombre, int pin, int saldo) {
        this.nombre = nombre;
        this.pin = pin;
        this.saldo = saldo;
    }
}

class Banco {
    List<Usuario> usuarios;

    Banco(List<Usuario> usuarios) {
        this.usuarios = usuarios;
    }

    boolean autenticar(String nombre, int pin) {
        for (Usuario usuario : usuarios) {
            if (usuario.nombre.equals(nombre)) {
                if (usuario.pin == pin) {
                    System.out.println("Estas Logeado");
                    return true;
                } else {
                    System.out.println("Pin o nombre incorrecto");
                    return false;
                }
            }
        }
        System.out.println("El usuario no existe");
        return false;
    }

    void sacarDinero(String nombre, int saldo) {
        for (Usuario usuario : usuarios) {
            if (usuario.nombre.equals(nombre)) {
                if (usuario.saldo < saldo) {
                    System.out.println("Saldo insuficiente");
                } else {
                    usuario.saldo -= saldo;
                    System.out.println("El saldo disponible es " + usuario.saldo);
                }
                break;
            }
        }
    }
}

public class cajero {
    public static void main(String[] args) {
        Usuario ana = new Usuario("Ana", 9872, 450);
        Usuario pablo = new Usuario("Pablo", 5555, 200);
        Usuario rodrigo = new Usuario("Rodrigo", 2222, 900);

        List<Usuario> usuarios = new ArrayList<>();
        usuarios.add(ana);
        usuarios.add(pablo);
        usuarios.add(rodrigo);

        Banco banco = new Banco(usuarios);
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("Bienvenido al Banco, por favor, identifiquese.");
            System.out.println("Introduzca el nombre.");
            String nombre = scanner.nextLine();
            System.out.println("Introduzca el pin");
            int pin = scanner.nextInt();
            scanner.nextLine(); // Limpiar el buffer

            if (banco.autenticar(nombre, pin)) {
                while (true) {
                    System.out.println("Por favor, elija una de estas opciones:\n 1. Sacar dinero\n 2. Terminar sesion.");
                    String opcion = scanner.nextLine();

                    if (opcion.equals("1")) {
                        System.out.println("Introduce cantidad a sacar.");
                        int saldo = scanner.nextInt();
                        scanner.nextLine(); // Limpiar el buffer
                        banco.sacarDinero(nombre, saldo);
                    } else if (opcion.equals("2")) {
                        System.out.println("Saliendo del sistema.");
                        break;
                    } else {
                        System.out.println("Opcion incorrecta. Por favor, introduzca otra opcion.");
                    }
                }
            } else {
                System.out.println("Usuario no autenticado. Por favor, Introduzca nombre y pin correctos.");
            }
        }
    }
}

File: Anyelo Carrillo Go/Cajero.go
package main

import (
	"fmt"
)

// Definición de la estructura Usuario
type Usuario struct {
	Nombre string
	Pin    int
	Saldo  float64
}

// Definición de la estructura Banco
type Banco struct {
	Usuarios []Usuario
}

// Método para autenticar un usuario
func (b *Banco) Autenticar(nombre string, pin int) bool {
	for _, usuario := range b.Usuarios {
		if usuario.Nombre == nombre {
			if usuario.Pin == pin {
				fmt.Println("Estás Logeado")
				return true
			} else {
				fmt.Println("Pin o nombre incorrecto")
				return false
			}
		}
	}
	fmt.Println("El usuario no existe")
	return false
}

// Método para sacar dinero
func (b *Banco) SacarDinero(nombre string, saldo float64) {
	for i, usuario := range b.Usuarios {
		if usuario.Nombre == nombre {
			if usuario.Saldo < saldo {
				fmt.Println("Saldo insuficiente")
				break
			} else if usuario.Saldo >= saldo {
				b.Usuarios[i].Saldo -= saldo
				fmt.Printf("El saldo disponible es %.2f\n", b.Usuarios[i].Saldo)
			}
		}
	}
}

func main() {
	// Crear usuarios
	ana := Usuario{Nombre: "Ana", Pin: 9872, Saldo: 450}
	pablo := Usuario{Nombre: "Pablo", Pin: 5555, Saldo: 200}
	rodrigo := Usuario{Nombre: "Rodrigo", Pin: 2222, Saldo: 900}

	// Crear banco con usuarios
	banco := Banco{Usuarios: []Usuario{ana, pablo, rodrigo}}

	// Bucle principal
	for {
		fmt.Println("Bienvenido al Banco, por favor, identifíquese.")
		fmt.Println("Introduzca el nombre.")
		var nombre string
		fmt.Scanln(&nombre)

		fmt.Println("Introduzca el pin")
		var pin int
		fmt.Scanln(&pin)

		if banco.Autenticar(nombre, pin) {
			for {
				fmt.Println("Por favor, elija una de estas opciones:\n 1. Sacar dinero\n 2. Terminar sesión.")
				var opcion string
				fmt.Scanln(&opcion)

				if opcion == "1" {
					fmt.Println("Introduce cantidad a sacar.")
					var saldo float64
					fmt.Scanln(&saldo)
					banco.SacarDinero(nombre, saldo)
				} else if opcion == "2" {
					fmt.Println("Saliendo del sistema.")
					break
				} else {
					fmt.Println("Opción incorrecta. Por favor, introduzca otra opción.")
				}
			}
		} else {
			fmt.Println("Usuario no autenticado. Por favor, introduzca nombre y pin correctos.")
		}
	}
}
